platform-core-module/src/main/java/com/softicar/platform/core/module/program/AGProgram.java

```diff
@@ -24,14 +30,54 @@ public AGProgram setProgramUuid(UUID programUuid) {
 		return setProgramUuid(AGUuid.getOrCreate(programUuid));
 	}
 
+	public DayTime getQueuedAt() {
+
+		return getOrCreateProgramState().getQueuedAt();
+	}
+
+	public void saveQueuedAt(DayTime value) {
+
+		getOrCreateProgramState().setQueuedAt(value).save();
+	}
+
+	public AGUser getQueuedBy() {
+
+		return getOrCreateProgramState().getQueuedBy();
+	}
+
+	public void saveQueuedBy(AGUser value) {
+
+		getOrCreateProgramState().setQueuedBy(value).save();
+	}
+
+	public Boolean isAbortRequested() {
+
+		return getOrCreateProgramState().isAbortRequested();
+	}
+
+	public void saveAbortRequested(Boolean value) {
```

Comment:
This method is only called only once, from `ProgramAbortRequester` - and it's unlikely that further references will be added. I think we should inline/remove it.

Reply:
I changed that.